pkg/syncer/k8scloudoperator/k8scloudoperator.go

```diff
@@ -312,6 +312,16 @@ func (k8sCloudOperator *k8sCloudOperator) PlacePersistenceVolumeClaim(ctx contex
 	}
 
 	spTypes, present := sc.Annotations[spTypeAnnotationKey]
+	//if affinity rule is specified then storage class should be of vsan-sna type
```

Comment:
This condition itself is correct now. In this final form, can we now combine it with the code below that already validates another condition when nodeAffinityAnnotationKey is present? Something like:
```
// if node affinity is specified on the PVC, then the storage class should be of vsan-sna type and WaitForFirstConsumer binding
...
```

Reply:
After offline discussion, we find that this entire feature of `node affinity on vSAN based on annotation` can be deprecated in favour of using the `vsan-sna StoragePool annotation` instead. Will can do that in a follow on change. So there is no point spending time optimizing this code that is going to be removed shortly. Please go ahead and merge this for now.

/approve